modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/ResourceManagementProviders.java

```diff
@@ -85,37 +83,18 @@ private Set<IResourceManagementProvider> getResourceManagementProviders(
     }
 
     private Set<IResourceManagementProvider> filterResourceManagementProviders(
-        MonitorConfiguration monitorConfig,
+        MonitorFilter filter,
         Set<IResourceManagementProvider> providers
     ) {
         Set<IResourceManagementProvider> providersToInclude = new HashSet<>();
 
-        // Find all the services that match any regex patterns in the 'includes' list
         for (IResourceManagementProvider provider : providers) {
-            for (Pattern includePattern : monitorConfig.getIncludesRegexList()) {
-                String providerName = provider.getClass().getCanonicalName();
-                Matcher includeMatcher = includePattern.matcher(providerName);
-                if (includeMatcher.matches()) {
-                    providersToInclude.add(provider);
-                    break;
-                }
-            }
-        }
-        
-        // From the filtered bundles, exclude any that match any regex patterns in the 'excludes' list
-        Set<IResourceManagementProvider> providersToExclude = new HashSet<>();
-        for (IResourceManagementProvider provider : providersToInclude) {
-            for (Pattern excludePattern : monitorConfig.getExcludesRegexList()) {
-                String providerName = provider.getClass().getCanonicalName();
-                Matcher excludeMatcher = excludePattern.matcher(providerName);
-                if (excludeMatcher.matches()) {
-                    providersToExclude.add(provider);
-                    break;
-                }
+            String monitorClassName = provider.getClass().getCanonicalName();
+            if (filter.isMonitorClassAllowed(monitorClassName)) {
```

Comment:
Method could be less specific. isPassesConstraints() ? Have a think and come up with something better ?

Reply:
Addressed in https://github.com/galasa-dev/galasa/pull/236/commits/a9ef197a47c1c7da97f4c1566f80ca24bce91184